### PR TITLE
feat: add seed list change hooks

### PR DESCRIPTION
**Pull request by Alex-A4**
## Status

**READY**

## Description

- Added tracking difference between `SeedList` changes

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/src/models/seed_changes/key_account_diffed.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import 'package:equatable/equatable.dart';
+
+import 'package:nekoton_repository/nekoton_repository.dart';
+
+/// [KeyAccountDiffed] is [KeyAccount] but without any actions.
+/// This model allows tracking KeyAccount adding/removing from [SeedList],
+/// see [SeedListDiffChange].
+class KeyAccountDiffed extends Equatable {
```
Comment (nesquikm) [inline lib/src/models/seed_changes/key_account_diffed.dart:8]: Maybe it's better to:
* rename this class `KeyAccountDiffed` -> `KeyAccountBase`
* Extend KeyAccount from this (`KeyAccountBase`) class. As I see it contains only these fields, but we can use EquatableMixin if it should compare its own fields fields.

copying code is bad :)

File: lib/src/models/seed_changes/seed_diffed.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import 'package:equatable/equatable.dart';
+import 'package:nekoton_repository/nekoton_repository.dart';
+
+/// [SeedDiffed] is just [Seed] where publicKey=masterKey and nothing more.
+/// This model allows tracking Seed adding/removing from [SeedList],
+/// see [SeedListDiffChange].
+class SeedDiffed extends Equatable {
```
Comment (nesquikm) [inline lib/src/models/seed_changes/seed_diffed.dart:7]: Same thing: `SeedDiffed` -> `SeedBase`, extend `Seed` from `SeedBase`?

File: lib/src/models/seed_changes/seed_key_diffed.dart [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import 'package:equatable/equatable.dart';
+
+import 'package:nekoton_repository/nekoton_repository.dart';
+
+/// [SeedKeyDiffed] is [SeedKey] (even master key=seed).
+/// This model allows tracking SeedKey adding/removing from [SeedList],
+/// see [SeedListDiffChange].
+class SeedKeyDiffed extends Equatable {
```
Comment (nesquikm) [inline lib/src/models/seed_changes/seed_key_diffed.dart:8]: Hey, it's really huge amount of copy and paste code isn't it? :)

File: lib/src/models/seed_changes/seed_list_diff_changes.dart [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+  /// 2) added/removed if its parent key added/removed
+  /// 3) added/removed if its seed added/removed
+  final List<KeyAccountDiffed> deletedAccounts;
+  final List<KeyAccountDiffed> addedAccounts;
+
+  /// Expand this diff by adding unique elements from [other].
+  SeedListDiffChange expand(SeedListDiffChange other) {
+    return SeedListDiffChange(
+      deletedSeeds:
+          <SeedDiffed>{...other.deletedSeeds, ...deletedSeeds}.toList(),
```
Comment (nesquikm) [inline lib/src/models/seed_changes/seed_list_diff_changes.dart:127]: Lets add comma before `}.toList` to make it uniform (like `deletedAccounts` and `addedAccounts`).
Now:
```dart
deletedSeeds:
          <SeedDiffed>{...other.deletedSeeds, ...deletedSeeds}.toList(),
```
With comma:
```dart
deletedSeeds:
          <SeedDiffed>{
            ...other.deletedSeeds,
            ...deletedSeeds,
          }.toList(),
```

File: lib/src/models/seed_changes/key_account_diffed.dart (hunk 1 above)
Comment (Alex-A4) [reply lib/src/models/seed_changes/key_account_diffed.dart]: -

File: lib/src/models/seed_changes/seed_diffed.dart (hunk 2 above)
Comment (Alex-A4) [reply lib/src/models/seed_changes/seed_diffed.dart]: -

File: lib/src/models/seed_changes/seed_key_diffed.dart (hunk 3 above)
Comment (Alex-A4) [reply lib/src/models/seed_changes/seed_key_diffed.dart]: -

File: lib/src/models/seed_changes/seed_list_diff_changes.dart (hunk 4 above)
Comment (Alex-A4) [reply lib/src/models/seed_changes/seed_list_diff_changes.dart]: -

**Review by nesquikm — APPROVED** [review]

**Pull request merged**